core/src/main/java/org/opensearch/sql/calcite/udf/udaf/ValuesAggFunction.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.calcite.udf.udaf;
+
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.TreeSet;
+import org.opensearch.sql.calcite.udf.UserDefinedAggFunction;
+
+/**
+ * VALUES aggregate function implementation. Returns distinct values from a field in lexicographical
+ * order as a multivalue field.
+ *
+ * <p>Behavior: - Returns unique values only (no duplicates) - Values are sorted in lexicographical
+ * order - Processes field values as strings (casts all inputs to strings) - No limit on number of
+ * values (unlike LIST which has 100 limit) - Supports only scalar data types (rejects STRUCT/ARRAY
+ * types) - Implementation uses TreeSet for automatic sorting and deduplication
+ */
+public class ValuesAggFunction
+    implements UserDefinedAggFunction<ValuesAggFunction.ValuesAccumulator> {
+
+  @Override
+  public ValuesAccumulator init() {
+    return new ValuesAccumulator();
+  }
+
+  @Override
+  public Object result(ValuesAccumulator accumulator) {
+    return accumulator.value();
+  }
+
+  @Override
+  public ValuesAccumulator add(ValuesAccumulator acc, Object... values) {
+    // Handle case where no values are passed
+    if (values == null || values.length == 0) {
+      return acc;
+    }
+
+    Object value = values[0];
+
+    // Filter out null values
+    if (value != null) {
+      // Convert value to string, handling all types safely
+      String stringValue = convertToString(value);
+      acc.add(stringValue);
+    }
+
+    return acc;
+  }
+
+  /** Converts any value to its string representation. */
+  private String convertToString(Object value) {
+    if (value == null) {
+      return null;
+    }
```

Comment:
`value` must be non-null here because of line 45?

Reply:
yes this is not necessary for removed it.